core/src/main/java/hudson/Proc.java

```diff
@@ -328,7 +328,16 @@ public int join() throws InterruptedException, IOException {
                     String msg = "Process leaked file descriptors. See http://wiki.jenkins-ci.org/display/JENKINS/Spawning+processes+from+build for more information";
                     Throwable e = new Exception().fillInStackTrace();
                     LOGGER.log(Level.WARNING,msg,e);
+                    if (LOGGER.isLoggable(Level.FINER)) {
+                        if (p != null) {
+                            LOGGER.log(Level.FINER, "Process with PID: " + p.getPid() + " leaked file descriptors.");
+                        } else {
+                            LOGGER.log(Level.FINER, "Process with no PID leaked file descriptors.");
```

Comment:
This message is quite confusing. I wonder if there is a way to construct something more helpful than a PID, we have the `name` at least...